mono/metadata/threads.c

```diff
@@ -1619,14 +1619,16 @@ mono_thread_set_name_internal (MonoInternalThread *this_obj, MonoString *name, g
 	else
 		this_obj->name = NULL;
 
-	
+	if (!(this_obj->state & ThreadState_Stopped))
+		tid = thread_get_tid (this_obj);
+
 	UNLOCK_THREAD (this_obj);
 
-	if (this_obj->name && this_obj->tid) {
+	if (this_obj->name && tid) {
```

Comment:
Would it be possible to have a check that does not rely on `tid != 0` since, in theory, `0` is a valid thread id on many systems.